src/server/pages/home-page.jsx

```diff
@@ -31,43 +121,163 @@ const LetsGetStarted = () => (
     <NewNotebookButton />
   </AttentionBlock>
 );
-export default class HomePage extends React.Component {
-  static propTypes = {
-    userInfo: PropTypes.shape({
-      name: PropTypes.string,
-      avatar: PropTypes.string
-    }),
-    headerMessage: PropTypes.string,
-    notebookList: PropTypes.arrayOf(PropTypes.object)
-  };
-  render() {
-    const isLoggedIn = "name" in this.props.userInfo;
-    const { notebookList } = this.props;
-    return (
-      <div>
-        <Header
-          userInfo={this.props.userInfo}
-          headerMessage={this.props.headerMessage}
-        />
-        <PageBody>
-          <TopContainer>
-            {!isLoggedIn && process.env.IODIDE_PUBLIC && (
-              <MarketingCopySplash />
-            )}
-            {!isLoggedIn && !process.env.IODIDE_PUBLIC && <LetsGetStarted />}
-            {isLoggedIn && <LoggedInSplash userInfo={this.props.userInfo} />}
-            <PageHeader>Try These Examples</PageHeader>
-            <FeaturedNotebooks width={`${sharedProperties.pageWidth}px`} />
-          </TopContainer>
-          <BelowFoldContainer>
-            {notebookList.length ? (
-              <TrendingNotebooksPage notebookList={notebookList} />
-            ) : (
-              <LetsGetStarted />
-            )}
-          </BelowFoldContainer>
-        </PageBody>
-      </div>
+
+let previousBodyOverflow;
```

Comment:
If I understand correctly, if it were in `HomePage`, it would be reset whenever the component is re-rendered. Some people use `useRef` for this.

https://stackoverflow.com/a/53351556/4297741

I'll double-check this, though, just to be safe.

Reply:
The homepage does re-render more than once while the overlay is open due to prop changes and we want to maintain a record of what the overlay style was before the overlay was opened up, so I'm going to leave this as-is.